spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleLauncher.java

```diff
@@ -87,6 +94,7 @@ public ModuleLauncher(ModuleResolver moduleResolver) {
 	 * @param aggregate whether the modules should be aggregated at launch
 	 * @param parentArgs a list of arguments for the whole aggregate
 	 */
+	@SuppressWarnings("unchecked")
 	public void launch(List<ModuleLaunchRequest> moduleLaunchRequests, boolean aggregate, Map<String,String> parentArgs) {
 		List<ModuleLaunchRequest> reversed = new ArrayList<>(moduleLaunchRequests);
 		Collections.reverse(reversed);
```

Comment:
The `reversed` ModuleLaunchRequests aren't used at all. I see this [change] (https://github.com/spring-cloud/spring-cloud-stream/commit/bd47560bd5af25b07c2bdba1f6ad7e33ee907da3#diff-3ca29f22b67f5c13875e045bd003bd6fL75) removed it. I think we need to use the `reversed` ModuleLaunchRequests when launching modules.  There is a catch here though: for the`direct-binding`, the `AggregateApplication` does the reverse when creating the module contexts.